source/develop/release-management/features/virt/parallel-migration-connections.md

```diff
@@ -0,0 +1,88 @@
+
+- It still applies that large guests with terrabytes of RAM are difficult to migrate successfully.
+
+- Since it's easier to saturate network bandwidth with multiple connections, it's advisable to not migrate multiple VMs from a single host at the same time.
+
+- VM CPU pinning may restrict the number of CPUs available to the QEMU process and thus impose a limit on the meaningful number of parallel connections.
+
+- It's difficult to guess the right number of parallel connections to use.  A rule of thumb is to use one connection per 10 Gbps of available network bandwidth.  If the number of connections is a bit high, some threads may remain idle while the related overhead should be negligible.
+
+- There is very little documentation about parallel migration connections and the information above was obtained directly from platform developers.
+
+## User experience
+
+A new migration option to enable parallel migration connections is introduced in cluster and VM edit dialogs.  The cluster option specifies what to use in the given cluster.  VMs can override the cluster value individually.  The new option offers the following choices:
+
+- **Disabled** (the default in clusters): Parallel migration connections are not used.
+
+- **Auto**: Use parallel migration connections (TODO: Even when there is insufficient network bandwidth, for the supposed benefit of a more efficient implementation?) and set their number automatically based on the available resources (network bandwidth and CPU power).
+
+- **Custom**: Use parallel migration connections and use the number of connections specified by the user.  The minimum number of connection is 2.  (TODO: Should we put any real limit on the maximum number?)
+
+- **Cluster value** (only in the VM edit dialog; it's the default there): Use parallel connections as specified in the VM's cluster settings. 
+
+We use **Disabled** as the cluster default because it's a newly introduced feature in oVirt and there is not enough experience with it.  We may consider changing it to **Auto** in next releases, especially if QEMU decides using parallel migration connections by default.
+
+If **Auto** is selected then the number of parallel connections is set to the lower number of the currently available migration network bandwidth divided by 10 Gbps and the number of VM vCPUs, but no less than 2.
+
+If parallel migration connections are enabled, the following adjustments are made when a migration is initiated:
+
+- Compression is disabled regardless of the selected migration policy.  (TODO: Check whether we can enable gzip/zstd compression.  If we can then it may be a good idea to always enable the compression.)
+
+- The maximum number of outgoing migrations is set to 1 regardless of the selected migration policy.  This may not be optimal if there is a non-parallel migration running that cannot saturate the available bandwidth but even in such a case it's generally simpler and safer not to start competing for the bandwidth.
```

Comment:
you mean when the cluster's setting changes, right? and what will happen - the engine will issue a call to VDSM in order to update the number of outgoing migrations to 1?

Reply:
I supposed it could be done in a similar way as with migration policies where we have different outgoing migrations limits for different policies. It's a migration parameter in the Vdsm API so no extra API call should be needed.

Reply:
ah ok, ack